frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/GitMacheteErrorReportSubmitter.java

```diff
@@ -113,12 +113,30 @@ private java.util.Map<String, String> getTemplateVariables(
           // This message is distinct from the throwable's message:
           // in `LOG.error(message, throwable)`, it's the first parameter.
           val messagePart = event.getMessage() != null ? (event.getMessage() + nl + nl) : "";
-          val throwablePart = event.getThrowableText().stripTrailing();
+          val throwablePart = shortenWrappedExceptionsStack(event.getThrowableText().stripTrailing());
           return "```${nl}${messagePart}${throwablePart}${nl}```";
         })
         .collect(Collectors.joining("${nl}${nl}"));
     templateVariables.put("stacktraces", stacktraces);
 
     return templateVariables;
   }
+
+  private String shortenWrappedExceptionsStack(String stackTrace) {
+    val nl = System.lineSeparator();
+    val rootCauseIndex = stackTrace.lastIndexOf("Caused by:");
```

Comment:
I'd also test against **suppressed exceptions**. Are they included in the stack trace by default? I'd also leave just first line of each suppressed exception's stack trace

Reply:
(can be done on a follow-up PR as this one is pretty self-contained & ready to merge)